pkg/odo/cli/url/create.go

```diff
@@ -284,8 +351,73 @@ func (o *URLCreateOptions) Run() (err error) {
 					break
 				}
 			}
+			err = o.EnvSpecificInfo.SetConfiguration("url", envinfo.EnvInfoURL{Name: o.urlName, Port: o.componentPort, ExposedPort: o.exposedPort, Kind: o.urlType})
+		} else {
+			containerEndpointMap, err := adapterutils.GetContainerEndpoints(o.devObj.Data)
+			if err != nil {
+				return errors.Wrap(err, "failed to get container endpoint map")
+			}
+			var firstContainer string
+			for containerName, endpointMap := range containerEndpointMap {
+				// save the first container name to store the new endpoint entry
+				if firstContainer == "" {
+					firstContainer = containerName
+				}
+				if _, exist := endpointMap[o.urlName]; exist {
+					if !o.forceFlag && !ui.Proceed(fmt.Sprintf("URL %v already exist in devfile endpoint entry under container %v. Do you want to override the endpoint", o.urlName, containerName)) {
+						log.Info("Aborted by the user")
+						return nil
+					}
+					if len(o.container) > 0 && o.container != containerName {
+						delete(endpointMap, o.urlName)
+					}
+					containerEndpointMap[containerName] = endpointMap
+				}
```

Comment:
I tried to create a URL with the same name and this part was not executed. It failed in the validate() part. Also since other commands don't support overriding in create, I think we can remove it from here.

Reply:
updated